zulipterminal/ui_tools/views.py

```diff
@@ -406,22 +406,34 @@ def __init__(self, width: int, view: Any) -> None:
             trcorner=u'─', blcorner=u'─', rline=u'',
             bline=u'─', brcorner=u'─'
             )
+        self.allow_update_user_list = True
         self.search_lock = threading.Lock()
         super(RightColumnView, self).__init__(self.users_view(),
                                               header=search_box)
 
     @asynch
-    def update_user_list(self, search_box: Any, new_text: str) -> None:
-        if not self.view.controller.editor_mode:
+    def update_user_list(self, search_box: Any=None,  # type: ignore
+                         new_text: str=None,
+                         user_list: Any=None) -> None:
+
+        assert ((user_list is None and search_box is not None and
+                 new_text is not None) or
+                (user_list is not None and search_box is None and
+                 new_text is None))
+
+        if not user_list and not self.view.controller.editor_mode:
             return
```

Comment:
Why is this `not self.view.controller.editor_mode`? From what I understand, this basically means that if the user is *not* searching, it'll return? I might be getting confused so please correct me if I'm wrong.

Reply:
Ah. Thanks for bringing this up @shreyamalviya ! This will definitely help pen down my though process behind the addition :+1: 
The `update_user_function` in itself gets called in 2 ways :+1: 
* 1: When the `urwid.connect` on self.user_search gets triggered  
* 2: Due to this new feature of update_user_list. 

When it gets triggered from part one, then we must return at `self.view.controller.edit_mode` as the user is not searching. (same reason you mentioned). However, If we call from the second position, we need to go ahead and update the user_list (rest of function). Therefore we check the extra condition here.